src/radiation_system.hpp

```diff
@@ -1206,18 +1198,32 @@ void RadSystem<problem_t>::AddSourceTerms(array_t &consVar, arrayconst_t &radEne
 		// Erad_guess is the new radiation energy (excluding work term)
 		// Egas_guess is the new gas internal energy
 
+		amrex::Real gas_update_factor = 1.0;
+		if (stage == 1) {
+			gas_update_factor = IMEX_a32;
+		}
+
 		// 2. Compute radiation flux update
 
 		amrex::GpuArray<amrex::Real, 3> Frad_t0{};
 		amrex::GpuArray<amrex::Real, 3> Frad_t1{};
 		amrex::GpuArray<amrex::Real, 3> dMomentum{0., 0., 0.};
+		std::array<double, 3> gasMtm{};
+		double realFourPiB = NAN;
 
 		T_gas = quokka::EOS<problem_t>::ComputeTgasFromEint(rho, Egas_guess, massScalars);
-		kappaFVec = ComputeFluxMeanOpacity(rho, T_gas);
+		kappaFVec = ComputeFluxMeanOpacity(rho, T_gas); // note the T_gas is allowed to be NAN
```

Comment:
Why is T_gas allowed to be NAN? That is not immediately obvious to me.

Reply:
Is this the case for the isothermal EOS?

Reply:
When gamma_ = 1.0, T_gas is NAN. In this case, we still need to do flux update via the backward Euler method and we need flux-mean opacity for that. The premise is that ComputeFluxMeanOpacity does not take T_gas as a useful parameter.